src/components/Resources/ResourceCard.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import styles from './Resource.module.css';
+
+function ResourceCard({ title, url, author, date, tags, tagMap }) {
+    // convert date into Object
+    const createdAt = new Date(date);
+
+    const convertedTag = (tags || []).map((id) => tagMap[id] || 'Unknown')
+
+  return (
+    <div className={styles.resource_card}>
+        <div className={styles.card_header}>
+            <h3>
+                <a href={url} target="_blank" rel="noopener noreferrer">
+                {title}
+                </a>
+            </h3>
+        </div>
+
+        <div className={styles.card_body}>
+            <p>
+                <strong> By: </strong>
+                <span className={styles.author}>{author} </span>
+            </p>
+            <p>
+                <strong> Published on: </strong>
+                <span className={styles.date}>
+                    {createdAt.toLocaleDateString('en-GB', {
+                        year: 'numeric',
+                        month: 'short',
+                        day: 'numeric',
+                    })}
+
+                </span>
+            </p>
+
+        </div>
+
+        <div className={styles.card_footer}>
+            {convertedTag && convertedTag.length > 0 && (
+                <p>Tags:
+                    {convertedTag.map((tag, index) => (
+                        <span className={styles.tags} key={index}>{tag}</span>
+                    ))}
+                </p>
+      )}
```

Comment:
Since the tags are inline `span` elements, [they wrap oddly and split when the cards get small.](https://i.imgur.com/2WuMEdC.png)

Instead, make the tags inline-blocks so they look the same even when they wrap:

```suggestion
            {convertedTag && convertedTag.length > 0 && (
              <div class={styles.tag_list}>
                <span>Tags:</span>
                {convertedTag.map((tag, index) => (<span className={styles.tag} key={index}>{tag}</span>))}
              </div>
            )}
```
[Here's what the fix looks like.](https://i.imgur.com/q2DMeF3.png)

The `tag_list` element exists only to ensure that the tags wrap with proper margin (through the `gap` property in CSS).

I've put the CSS for this as a suggestion in `Resource.module.css`.